pkg/apis/portworx/v1/portworxdiag.go

```diff
@@ -0,0 +1,101 @@
+package v1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// PortworxDiagResourceName is name for "portworxdiag" resource.
+	PortworxDiagResourceName = "portworxdiag"
+	// PortworxDiagResourcePlural is plural for "portworxdiag" resource.
+	PortworxDiagResourcePlural = "portworxdiags"
+)
+
+// PortworxDiagSpec is the spec used to define a portworx diag.
+type PortworxDiagSpec struct {
+	// Configuration for diags collection of the main Portworx component.
+	Portworx *PortworxComponent `json:"portworx,omitempty"`
```

Comment:
in the design spec, we also discussed about these flags:
  collectK8SObjects: true | false
  collectPodLogs: true | false
  cloudUpload: true | false

Reply:
That's correct and we will add those flag when we implement those parts.
As described in the [design doc for pxone](https://docs.google.com/document/d/1vWrujUfycoOJ2BAeeDoHiSrhVDRneHRCTBpgNOzYji8/edit#), we are only going to collect/upload what we have in porx and not adding anything new due to time constraints.

As described in your broader cluster wide diags doc, we will start adding support for the other options later.